### PR TITLE
Task #215728 : Complete UI Student basic details

### PR DESCRIPTION
**Pull request by rahulg1254**
Student Basic Details View
Task Link : https://tracker.tekdi.net/issues/215728
basicD2

## Summary by CodeRabbit

- **New Features**
	- Added a `StudentDetails` page showcasing student information like date of birth, education, grade, attendance, and test reports.
	- Implemented lazy loading for the `StudentBasicDetails` page to enhance performance.

- **Style**
	- Enhanced styling for the `StudentStatsCard` component to improve visual consistency.

### CODE REVIEW

File: src/pages/StudentBasicDetails.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        <CardContent>
+          <Box
+            sx={{
+              display: "flex",
+              alignItems: "center",
+              justifyContent: "space-between",
+            }}
+          >
+            <Typography
+              sx={{ fontSize: "16px", fontWeight: 600, color: "#4D4639" }}
```
Comment (itsvick) [inline src/pages/StudentBasicDetails.tsx:38]: @rahulg1254 get color from the themes file

File: src/pages/StudentBasicDetails.tsx [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+            marginTop: "20px",
+            height: "688px",
+            width: "340px",
+            margin: "auto",
+            borderRadius: "16px",
+            boxShadow: "none",
+          }}
+        >
+          {studentData.map((item, index) => (
+            <Box key={index} sx={{ padding: 2 }}>
```
Comment (itsvick) [inline src/pages/StudentBasicDetails.tsx:156]: @rahulg1254 dont use index as key

File: src/routes/routes.tsx [hunk 3]
```diff
@@ -13,6 +14,7 @@ function AllRoutes(): JSX.Element {
       <Suspense fallback={<TextField>{t("COMMON.LOADING")}</TextField>}>
         <Routes>
           <Route path="/" element={<LoginPage />} />
+          <Route path="/basicdetails" element={<StudentDetails />} />
```
Comment (itsvick) [inline src/routes/routes.tsx:17]: @rahulg1254 change route to /student-details

Comment (rahulg1254) [reply src/routes/routes.tsx:17]: changed the route

File: src/pages/StudentBasicDetails.tsx (hunk 2 above)
Comment (rahulg1254) [reply src/pages/StudentBasicDetails.tsx:156]: added item.id as a key

File: src/pages/StudentBasicDetails.tsx (hunk 1 above)
Comment (rahulg1254) [reply src/pages/StudentBasicDetails.tsx]: used customStyles common theme for colors

**Pull request merged**